contributed_definitions/NXem_base.nxdl.xml

```diff
@@ -48,49 +48,42 @@ template to be used for an application definition
          to explore physical mechanisms and phenomena, or the desire to characterize materials
          using electron microscopy.
     </doc>
-    <group type="NXprogram">
-        <doc>
-             A collection of all programs and libraries which are considered relevant
-             to understand with which software tools this NeXus file instance was
-             generated. Ideally, to enable a binary recreation from the input data.
-             
-             Examples include the name and version of the libraries used to write the
-             instance. Ideally, the software that writes these :ref:`NXprogram` instances
-             also includes the version of the set of NeXus classes i.e. the specific
-             set of base classes, application definitions, and contributed definitions
-             with which the here described concepts can be resolved.
-             
-             For the `pynxtools library &lt;https://github.com/FAIRmat-NFDI/pynxtools&gt;`_
-             which is used by the `NOMAD &lt;https://nomad-lab.eu/nomad-lab&gt;`_
-             research data management system, it makes sense to store e.g. the GitHub
-             repository commit and respective submodule references used.
-        </doc>
-    </group>
     <!--each NeXus file instance should have a default plot
 however as there are cases when this cannot be assured we cannot
 make the default required, one example is e.g. a NeXus instance
 where scientists just store conventions without a default plot-->
-    <group name="cs_profiling" type="NXcs_profiling">
-        <doc>
-             The configuration of the I/O writer software (e.g. `pynxtools &lt;https://github.com/FAIRmat-NFDI/pynxtools&gt;`_)
-             which was used to generate this NeXus file instance.
-        </doc>
-    </group>
     <group type="NXentry">
-        <attribute name="version" type="NX_CHAR">
-            <doc>
-                 An at least as strong as SHA256 hashvalue of the file
-                 which specifies the application definition.
-            </doc>
-        </attribute>
         <field name="definition" type="NX_CHAR">
+            <attribute name="version" type="NX_CHAR">
+                <enumeration>
+                    <item value="NXem"/>
+                </enumeration>
+            </attribute>
+        </field>
+        <group name="cs_profiling" type="NXcs_profiling">
             <doc>
-                 NeXus NXDL schema to which this file conforms.
+                 The configuration of the I/O writer software (e.g. `pynxtools &lt;https://github.com/FAIRmat-NFDI/pynxtools&gt;`_)
+                 which was used to generate this NeXus file instance.
             </doc>
-            <enumeration>
-                <item value="NXem"/>
-            </enumeration>
-        </field>
+            <group type="NXprogram">
```

Comment:
should not it be collected in an NXprocess, e.g. data_transfer(NXprocess) or data_generation(NXprocess)? Is not it redundant with NXroot attributes?

Reply:
No it is not redundant but maybe even a proposal how to improve NXroot.
The idea of NXcs_profiling is to store benchmarking and performance measurements or give details (like I do for paraprobe-toolbox) about the computer, in appdefs I slimmed down the description of a computer but the base classes remain to describe every tiny detail even of the hardware components of which that computer is built.

Reply:
On the relative location of the profiling group yeah my understanding is that if I attach it to a child somewhere in the tree one may conclude that implicitly this means these profiling data were collected only for the process in which that group is located. Here I placed it on top but yeah the relative location is a choice for which I can find several locations with substantiated arguments.

Reply:
NXcs_profiling does not contain NXprogram (should not it be NXsoftware?) but only this specialisation here /NXem_base/ENTRY/cs_profiling. Why the base class does not reference the software to be profiled?